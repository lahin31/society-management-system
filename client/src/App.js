import React, { useState, useEffect } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';

import AuthContext from './contexts/auth-context';

import IndexPage from './pages/Index';
import HomePage from './pages/home/Home';
import Login from './pages/authentications/Login';
import Registration from './pages/authentications/Registration';
import ForgetPassword from './pages/authentications/ForgetPassword';
import ConfirmingAccount from './pages/authentications/ConfirmingAccount';
import ProfilePage from './pages/user/Profile';
import EditProfile from './pages/user/EditProfile';
import ContactUs from './pages/contact-us/ContactUs';
import AboutUs from './pages/about-us/AboutUs';
import ResetPassword from './pages/authentications/ResetPassword';
import Search from './pages/search/Search';

import Navbar from './components/navbar/Navbar';

import './App.scss';
import SocietyDetails from './pages/societies/society/SocietyDetails';

function App() {
  const [token, setToken] = useState('');
  const [, setRefreshToken] = useState('');
  const [userId, setUserId] = useState('');
  const [authenticateUser] = useState({});
  const [tokenExpiration, setTokenExpiration] = useState('');

  useEffect(() => {
    const accessToken = JSON.parse(
      localStorage.getItem('accessToken')
    );
    const userIdLocal = JSON.parse(localStorage.getItem('userId'));
    const tokenExp = JSON.parse(
      localStorage.getItem('tokenExpiration')
    );
    if (accessToken && userIdLocal && tokenExp) {
      setToken(accessToken);
      setUserId(userIdLocal);
      setTokenExpiration(tokenExp);
    }
  }, []);

  const login = (token, refreshToken, userId, tokenExpiratopn) => {
    setToken(token);
    setRefreshToken(refreshToken);
    setUserId(userId);
    setTokenExpiration(tokenExpiratopn);
  };

  const logout = () => {
    setToken(null);
    setRefreshToken(null);
    setUserId(null);
    setTokenExpiration(null);
    localStorage.removeItem('accessToken');
    localStorage.removeItem('refreshToken');
    localStorage.removeItem('userId');
    localStorage.removeItem('tokenExpiration');
  };

  return (
    <Router>
      <AuthContext.Provider
        value={{
          token,
          userId,
          authenticateUser,
          tokenExpiration,
          login,
          logout,
        }}
      >
        <main>
          <Navbar />
          <Switch>
            {!token && <Route path="/" exact component={IndexPage} />}
            {token && <Route path="/" exact component={HomePage} />}
            {token && (
              <Route
                path="/profile/:username"
                component={ProfilePage}
              />
            )}
            {token && (
              <Route
                path="/edit-profile/:username"
                component={EditProfile}
              />
            )}
            {token && <Route path="/search/:searchVal" component={Search} />}
            {token && <Route path="/societies/:society_id" component={SocietyDetails} />}
            {token && <Route path="/about_us" component={AboutUs} />}
            {token && <Route path="/contact_us" component={ContactUs} />}
            {!token && <Route path="/login" component={Login} />}
            {!token && (
              <Route path="/registration" component={Registration} />
            )}
            {!token && (
              <Route
                path="/forget-password"
                component={ForgetPassword}
              />
            )}
            {!token && (
              <Route
                path="/confirmation/:token"
                component={ConfirmingAccount}
              />
            )}
            {!token && (
              <Route path="/reset/:token" component={ResetPassword} />
            )}
          </Switch>
        </main>
      </AuthContext.Provider>
    </Router>
  );
}

export default App;
