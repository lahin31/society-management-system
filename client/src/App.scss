@import "./constants/global/variables.scss";

.makeStyles-paper-33,
.makeStyles-paper-266 {
  margin-top: 40px !important;
}

.cm_wrapper {
  display: flex;
  justify-content: space-between;
  margin-top: 60px;
}

.nav_btn {
  margin-right: 50px !important;
  margin-top: 20px !important;
  color: #fff !important;
  background-color: var(--color-primary) !important;
}

.fa-arrow-right {
  background: #fff;
  padding: 6px;
  color: var(--color-primary);
  margin-left: 7px;
  border-radius: 50%;
}

.body_logo_img {
  position: absolute;
  top: 200px;
  text-align: right;
  pointer-events: none;
}

.index_page_svg {
  margin-top: 263px;
  width: 100%;
}

#body_logo_img {
  width: 45%;
  padding-right: 0;
}

.body_text {
  position: absolute;
  top: 140px;
}

.body_text_title {
  font-size: 48px;
  font-weight: 700;
  padding-left: 45px;
  color: #32353E;
}

a {
  text-decoration: none;
}

.error_msg_wrap,
.success_msg_wrap {
  margin-bottom: 12px;
}

.info {
  text-align: center;
}

/* simple - enter transition 300ms, exit 150ms */
.fade-appear,
.fade-enter {
  opacity: 0;
  z-index: 1;
}

.fade-appear-active,
.fade-enter.fade-enter-active {
  opacity: 1;
  transition: opacity 300ms linear 150ms;
}

.fade-exit {
  opacity: 1;
}

.fade-exit.fade-exit-active {
  opacity: 0;
  transition: opacity 150ms linear;
}

form {
  .submit_btn {
    background-color: var(--color-primary);
    color: #fff;
  }

  .submit_btn:hover {
    color: #32353E;
  }
}


@media(max-width: 850px) {
  .body_text {
    width: 100%;
  }

  .body_text_title {
    font-size: 23px;
  }

  .body_logo_img {
    top: 300px;
    text-align: center;
  }
}