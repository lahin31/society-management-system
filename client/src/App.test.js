/* eslint-disable */
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

afterEach(cleanup);

// making a snapshot
it('renders', () => {
  const {asFragment} = render( <App /> );
  expect(asFragment()).toMatchSnapshot();
});

/**
 * renderWithRouter(), to render the component(App) and pass history to the Router component.
 * this function will help us to know if the initial page is index or not
 */
const renderWithRouter = component => {
  const history = createMemoryHistory(); // createMemoryHistory(), to create a navigation history
  return {
    ...render(<Router history={history}>{component}</Router>),
  }
}

it("should render the index page", () => {
  const {getByTestId} = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const link = getByTestId("home-link");
  expect(navbar).toContainElement(link);
});

it("should navigate to the login page", () => {
  const {getByTestId} = renderWithRouter(<App />);
  fireEvent.click(getByTestId("login-link"));
  expect(getByTestId("login-title")).toHaveTextContent("Login")
});