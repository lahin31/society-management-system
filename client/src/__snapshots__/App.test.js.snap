// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <main>
    <nav
      class="navbar navbar-success"
      data-testid="navbar"
      style="justify-content: space-between;"
    >
      <a
        data-testid="home-link"
        href="/"
        id="logo"
      >
        Society Management System
      </a>
      <a
        data-testid="login-link"
        href="/login"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained nav_btn MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </nav>
    <svg
      class="index_page_svg"
      viewBox="0 0 1440 320"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0,32L24,53.3C48,75,96,117,144,144C192,171,240,181,288,176C336,171,384,149,432,138.7C480,128,528,128,576,149.3C624,171,672,213,720,202.7C768,192,816,128,864,112C912,96,960,128,1008,165.3C1056,203,1104,245,1152,229.3C1200,213,1248,139,1296,112C1344,85,1392,107,1416,117.3L1440,128L1440,320L1416,320C1392,320,1344,320,1296,320C1248,320,1200,320,1152,320C1104,320,1056,320,1008,320C960,320,912,320,864,320C816,320,768,320,720,320C672,320,624,320,576,320C528,320,480,320,432,320C384,320,336,320,288,320C240,320,192,320,144,320C96,320,48,320,24,320L0,320Z"
        fill="#EAF4E9"
        fill-opacity="1"
      />
    </svg>
    <div
      class="cm_wrapper"
    >
      <div
        class="body_text"
      >
        <h1
          class="body_text_title"
        >
          Society Management System
        </h1>
        <p>
          We help you to engage with your societies
        </p>
        <div
          class="action_btn text-align-center"
        >
          <a
            href="/login"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained nav_btn MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Join Today 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
      <div
        class="body_logo_img"
      >
        <img
          alt="body_logo"
          id="body_logo_img"
          src="bodylogo.png"
        />
      </div>
    </div>
    <div
      class="info"
    >
      <div
        class="row"
      >
        <div
          class="column"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5c-3.87 0-7 3.13-7 7h2c0-2.76 2.24-5 5-5s5 2.24 5 5h2c0-3.87-3.13-7-7-7zm1 9.29c.88-.39 1.5-1.26 1.5-2.29 0-1.38-1.12-2.5-2.5-2.5S9.5 10.62 9.5 12c0 1.02.62 1.9 1.5 2.29v3.3L7.59 21 9 22.41l3-3 3 3L16.41 21 13 17.59v-3.3zM12 1C5.93 1 1 5.93 1 12h2c0-4.97 4.03-9 9-9s9 4.03 9 9h2c0-6.07-4.93-11-11-11z"
            />
          </svg>
          <h2>
            Connect
          </h2>
          <p>
            Lorem ipsum dolor sit amet, Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet 
          </p>
        </div>
        <div
          class="column"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 1H7c-1.1 0-2 .9-2 2v18c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 18H7V5h10v14zM12 6.72c-1.96 0-3.5 1.52-3.5 3.47h1.75c0-.93.82-1.75 1.75-1.75s1.75.82 1.75 1.75c0 1.75-2.63 1.57-2.63 4.45h1.76c0-1.96 2.62-2.19 2.62-4.45 0-1.96-1.54-3.47-3.5-3.47zm-.88 8.8h1.76v1.76h-1.76z"
            />
          </svg>
          <h2>
            Engage
          </h2>
          <p>
            Lorem ipsum dolor sit amet, Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet 
          </p>
        </div>
        <div
          class="column"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 20c-.29 0-.56-.06-.76-.15-.71-.37-1.21-.88-1.71-2.38-.51-1.56-1.47-2.29-2.39-3-.79-.61-1.61-1.24-2.32-2.53C9.29 10.98 9 9.93 9 9c0-2.8 2.2-5 5-5s5 2.2 5 5h2c0-3.93-3.07-7-7-7S7 5.07 7 9c0 1.26.38 2.65 1.07 3.9.91 1.65 1.98 2.48 2.85 3.15.81.62 1.39 1.07 1.71 2.05.6 1.82 1.37 2.84 2.73 3.55.51.23 1.07.35 1.64.35 2.21 0 4-1.79 4-4h-2c0 1.1-.9 2-2 2zM7.64 2.64L6.22 1.22C4.23 3.21 3 5.96 3 9s1.23 5.79 3.22 7.78l1.41-1.41C6.01 13.74 5 11.49 5 9s1.01-4.74 2.64-6.36zM11.5 9c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5-1.12-2.5-2.5-2.5-2.5 1.12-2.5 2.5z"
            />
          </svg>
          <h2>
            Know
          </h2>
          <p>
            Lorem ipsum dolor sit amet, Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet 
          </p>
        </div>
      </div>
    </div>
    <div
      class="footer_wrapper"
      style="margin-top: 20px;"
    >
      <p
        class="text-align-center font-weight-bold"
      >
        Â© 2020 Society Management System, Inc.
      </p>
    </div>
  </main>
</DocumentFragment>
`;
